api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -410,4 +410,18 @@ default UpdateSchema updateColumn(String name, Type.PrimitiveType newType, Strin
   default UpdateSchema setIdentifierFields(String... names) {
     return setIdentifierFields(Sets.newHashSet(names));
   }
+
+  /**
+   * Determines if the case of schema needs to be considered when updating schema
+   *
+   * @param caseSensitive when false, the case of fields in schema is ignored
+   * @return this for method chaining
+   * @throws IllegalStateException If it encounters errors during provided schema traversal
+   * @throws IllegalArgumentException If name doesn't identify a column in the schema or if this
+   *     change introduces a type incompatibility or if it conflicts with other additions, renames,
+   *     or updates.
+   */
+  default UpdateSchema caseSensitive(boolean caseSensitive) {
+    throw new UnsupportedOperationException();
```

Comment:
Ok one issue here, We only apply the implementation of "caseSensitive" to  unionXXX but I think it also applies to

```
updateColumn
updateColumnDoc
makeColumnOptional
requireColumn
deleteColumn
moveFirst
moveBefore (2x here)
moveAfter (2x here as well)
```

Reply:
I have made the code change. I will add tests for these change